src/include/distributed/multi_copy.h

```diff
@@ -43,8 +47,54 @@ typedef struct NodeAddress
 	int32 nodePort;
 } NodeAddress;
 
+/* CopyDestReceiver can be used to stream results into a distributed table */
+typedef struct CitusCopyDestReceiver
+{
+	DestReceiver pub;
+
+	/* relation and columns to which to copy */
```

Comment:
Is there any other pre-existing struct we can slap in here which encapsulates much of the following? It seems as though we're cramming basically the entire denormalized set of distribution metadata for a table in here…